src/main/java/core/basesyntax/converter/DataConverterImpl.java

```
@@ -0,0 +1,32 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> readFile) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String report : readFile) {
+            String[] reportSplit = report.split(",");
+            String code = reportSplit[0];
+            String fruit = reportSplit[1];
+            int quantity = Integer.parseInt(reportSplit[2]);
+            FruitTransaction.Operation operationCode = null;
+            for (FruitTransaction.Operation operation
+                    : FruitTransaction.Operation.values()) {
+                if (operation.getCode().equals(code)) {
+                    operationCode = operation;
+                    break;
+                }
+            }
+            if (operationCode != null) {
```

Silently ignoring lines with an invalid operation code can hide errors in the input file. According to the task requirements, you need to handle invalid data. It is better to fail fast when encountering data you don't know how to process.

If no matching operation is found, you should throw an exception (e.g., `IllegalArgumentException`) to indicate that the input file is malformed. This provides immediate feedback about the problem.